### PR TITLE
core: add FromJSON instance to ChatResponse

### PR DESCRIPTION
**Pull request by dpwiz**

### CODE REVIEW

File: src/Simplex/Chat/Controller.hs [hunk 1]
```diff
@@ -438,10 +445,10 @@ data ChatCommand
...
   = CRActiveUser {user :: User}
   | CRUsersList {users :: [UserInfo]}
-  | CRChatStarted
-  | CRChatRunning
-  | CRChatStopped
-  | CRChatSuspended
+  | CRChatStarted {_nullary :: Maybe Int}
+  | CRChatRunning {_nullary :: Maybe Int}
+  | CRChatStopped {_nullary :: Maybe Int}
+  | CRChatSuspended {_nullary :: Maybe Int}
```
Comment (dpwiz) [inline src/Simplex/Chat/Controller.hs:451]: `Maybe Void`?

File: src/Simplex/Chat/Messages/CIContent.hs [hunk 2]
```diff
@@ -722,3 +696,31 @@ ciCallInfoText status duration = case status of
   CISCallProgress -> "in progress " <> durationText duration
   CISCallEnded -> "ended " <> durationText duration
   CISCallError -> "error"
+
+$(JQ.deriveJSON (enumJSON $ dropPrefix "CISCall") ''CICallStatus)
```
Comment (dpwiz) [inline src/Simplex/Chat/Messages/CIContent.hs:700]: No need for ()s, just `$JQ.derive ...`

Comment (epoberezkin) [reply src/Simplex/Chat/Messages/CIContent.hs:700]: It's a common style I think

File: src/Simplex/Chat/Controller.hs (hunk 1 above)
Comment (dpwiz) [reply src/Simplex/Chat/Controller.hs:451]: It wouldn't. Void makes `Just` unrepresentable, making Nothing the only option (to be eliminated later).

File: src/Simplex/Chat/Types.hs [hunk 3]
```diff
@@ -1550,6 +1574,24 @@ instance ToJSON ChatVersionRange where
...
+
+$(JQ.deriveJSON defOpts ''User)
+
+$(JQ.deriveJSON defOpts ''NewUser)
+
+$(JQ.deriveJSON defOpts ''UserInfo)
+
+$(JQ.deriveJSON defOpts ''Contact)
+
+$(JQ.deriveJSON defOpts ''ContactRef)
```
Comment (dpwiz) [inline src/Simplex/Chat/Types.hs:1597]: Missing:
```
$(JQ.deriveJSON defOpts ''Notification)
```

**Pull request merged**